packages/installer/upgrade.sh

```diff
@@ -84,50 +56,55 @@ find_upgrade_channel() {
 }
 
 mount_image() {
+    STATEDIR=/run/initramfs/isoscan
     TARGET=/tmp/upgrade
-    mkdir ${TARGET} || true
-    mount $TARGET_PARTITION ${TARGET}
+
+    mkdir -p $TARGET || true
+    # mkdir -p $STATEDIR || true
+    # mount ${STATE} $STATEDIR
+    mount -o remount,rw ${STATE} ${STATEDIR}
+
+    mkdir -p ${STATEDIR}/cOS || true
+    rm -rf ${STATEDIR}/cOS/transition.img || true
+    dd if=/dev/zero of=${STATEDIR}/cOS/transition.img bs=1M count=3240
+    mkfs.ext4 ${STATEDIR}/cOS/transition.img
+    mount -t ext4 -o loop ${STATEDIR}/cOS/transition.img $TARGET
 }
 
 mount_persistent() {
     mkdir -p ${TARGET}/oem || true
     mount ${OEM} ${TARGET}/oem
     mkdir -p ${TARGET}/usr/local || true
     mount ${PERSISTENT} ${TARGET}/usr/local
-    GRUB=$(blkid -L COS_GRUB || true)
-    if [ -n "$GRUB" ]; then
-        mkdir -p ${TARGET}/boot/efi || true
-        mount ${GRUB} ${TARGET}/boot/efi
-    fi
 }
 
 upgrade() {
     mount_image
-
-    # XXX: Wipe old, needed until we have a persistent luet state.
-    # TODO: at least cache downloads before wiping and we are sure we can perform the new install
-    if [ -d "/tmp/empty" ]; then
-        rm -rf /tmp/empty
-    fi
-    mkdir /tmp/empty
-    rsync -a --delete /tmp/empty/ /tmp/upgrade/
-
     mount_persistent
     ensure_dir_structure
+
+    mkdir -p /run/tmp
     # FIXME: XDG_RUNTIME_DIR is for containerd, by default that points to /run/user/<uid>
     # which might not be sufficient to unpack images. Use /usr/local/tmp until we get a separate partition
     # for the state
     # FIXME: Define default /var/tmp as tmpdir_base in default luet config file
-    XDG_RUNTIME_DIR=/var/tmp TMPDIR=/var/tmp luet install -y $UPGRADE_IMAGE
+    XDG_RUNTIME_DIR=/run/tmp TMPDIR=/run/tmp luet install -y $UPGRADE_IMAGE
```

Comment:
This is tricky. Probably the best would be to use a temporary `/usr/local/tmp`

Reply:
moved do `/usr/local/tmp` in https://github.com/mudler/cOS/pull/58/commits/f2048686547e6e23f6e0a87671236cfd1c20215a